### ISSUE TITLE
JDK-8088418: Reintroduce JFR Pulse Logger

### ISSUE THREAD

**Issue by thegreystone** [javafxports/openjdk-jfx#261]
In JDK 8, there was a JFR based pulse logger available. Now that JFR and JFX both are open sourced, we should reintroduce it, using the open JFR APIs.

I'd be happy to create a pull request for this.

**Comment by kevinrushforth** [comment]
Sure, that would be great.

Please also file a JBS bug per the instructions in [CONTRIBUTING.md](https://github.com/javafxports/openjdk-jfx/blob/develop/.github/CONTRIBUTING.md) and the [code review policies](https://wiki.openjdk.java.net/display/OpenJFX/Code+Reviews).

**Comment by kevinrushforth** [comment]
@thegreystone  As far as implementation, I would prefer not to introduce a hard-dependency at build time (and certainly not at runtime). If it could be done via reflection then that would be one way to meet this goal. What were you thinking of proposing?

**Comment by kevinrushforth** [comment]
We might as well repurpose [JDK-8088418](https://bugs.openjdk.java.net/browse/JDK-8088418), which you filed back in the JDK 9 time frame, to track this work.

**Comment by kevinrushforth** [comment]
Regarding the earlier comment I wrote:

> I would prefer not to introduce a hard-dependency at build time (and certainly not at runtime)...

I think this is based on my misconception. If the JFR APIs that you propose to use are part of the JDK, then there is no problem using them directly.

**Comment by thegreystone** [comment]
Perhaps something like this? #264

**Comment by kevinrushforth** [comment]
I'll take a look at it when I get a chance. Thanks.

**Comment by kevinrushforth** [comment]
I don't see much value in maintaining compatibility with events in JDK 8, so whichever way you want to go is fine. I'll take a quick look at this in the next day or two (I hope). One thing to note is that you have white-space problems as shown by the Travis CI failure. Also, if you could revert the unrelated formatting changes to the existing code it would make it easier to review.

**Comment by thegreystone** [comment]
Where can I see the white space problems? 

I see a lot of these:
test.com.sun.javafx.binding.BidirectionalBindingTest > testBrokenBind[0] STANDARD_ERROR
    Exception in thread "Test worker" java.lang.RuntimeException: Bidirectional binding failed, setting to the previous value
    	at javafx.base/com.sun.javafx.binding.BidirectionalBinding$BidirectionalBooleanBinding.changed(BidirectionalBinding.java:284)

**Comment by kevinrushforth** [comment]
Those are all informational log messages, and are fine. You'll see the error at the end of the log for Linux or Mac:

```
--- Checking for white space errors

modules/javafx.base/src/main/java/com/sun/javafx/logging/PulseLogger.java :trailingWhitespace:
modules/javafx.base/src/main/java/com/sun/javafx/logging/jfr/JFRPulseLogger.java :trailingWhitespace:

Found 2 whitespace or executable issues

To correct, use
   bash tools/scripts/checkWhiteSpace -F -x

Error in white space check

travis_time:end:01d0036a:start=1541588162552012733,finish=1541589002578492238,duration=840026479505
[0K
[31;1mThe command "bash .ci/script.sh" exited with 1.
```

**Comment by kevinrushforth** [comment]
Fixed by #272
